### ISSUE TITLE
Detect `unittest.fail()` calls for assignment warning

### ISSUE THREAD

**Issue by Flamefire** [pylint-dev/pylint#10743]
### Current problem

See this example
```
from unittest import TestCase
import os

class TestX(TestCase):
  def test_foo(self):
    if 'FOO' in os.environ:
        x = 1
    else:
        self.fail()
    print(x) # E0606:possibly-used-before-assignment
```

Although there is no way to reach the last line without `x` being unassigned it is reported as such.

### Desired solution

`self.fail()` should be detected as terminating similar to `raise ...`

**Comment by CodePapayas** [comment]
Hello, may I please be assigned this issue? I would like to give it a shot.

Thanks!

**Comment by Pierre-Sassoulas** [comment]
Sure, thank you for contributing !
